tests/entities/test_listeners.py

```diff
@@ -37,6 +39,14 @@ class TestListeners:
         modified_by="test_user_id",
     )
 
+    @pytest.fixture(autouse=True)
+    def setup(self):
+        self.fi_history.reset_mock()
+        self.fi_history.columns = {"name": "test"}
+        self.mapping_history.reset_mock()
+        self.mapper.reset_mock()
+        self.connection.reset_mock()
+
```

Comment:
Just making sure I understand this setup, because you have the mocked tables at the test class level, and are reused between test functions to setup the listener function, you're just resetting any data in them.  Correct?

Reply:
exactly, so in between tests the mocks don't carry the actions that went against them; don't think there was a initMocks type of thing here like in the java world, so need to reset them between the tests.

Reply:
Cool was not aware of that reset_mock(), good to know for future.